### PR TITLE
REL: update Paver file to ensure sdist contents are OK for releases.

### PR DESCRIPTION
**Pull request by rgommers**
Closes gh-5195 (missing Sphinx theme for 0.16.0 tarballs).

Also remove `release.sh` - we're not using it anymore.  Instead, we'll be using `numpy-vendor/fabfile.py`

[ci skip]

### CODE REVIEW

File: pavement.py [hunk 1]
```diff
@@ -310,8 +310,16 @@ def tarball_name(type='gztar'):
 
 @task
 def sdist():
-    # To be sure to bypass paver when building sdist... paver + scipy.distutils
+    # First clean the repo and update submodules (for up-to-date doc html theme
+    # and Sphinx extensions)
+    sh('git clean -xdf')
```
Comment (charris) [inline pavement.py]: Should probably do this for numpy also.

Comment (rgommers) [inline pavement.py]: done in https://github.com/numpy/numpy/pull/6474

**Comment by rgommers** [comment]
Alright, the identical numpy change is merged and I need to backport this, so in it goes.

**Comment by rgommers** [comment]
backported in e437297

**Pull request merged**